libmattermost.c

```diff
@@ -108,6 +108,22 @@ json_array_to_string(JsonArray *array)
 	return str;
 }
 
+// Mattermost uses unnamed arrays that json-glib parser turns into strings ...
+JsonArray *
+json_named_array_from_string(const gchar *name, const gchar *str)
+{
+	JsonArray *array = NULL;
+	JsonParser *parser = json_parser_new();
+	gchar *data = g_strconcat("{\"", name , "\":", str, "}", NULL);
+

```

Comment:
In that case, can you provide an example of what kind of invalid JSON this function is trying to fix?  Otherwise it just looks like this function is doing something wrong without some kind of explanation